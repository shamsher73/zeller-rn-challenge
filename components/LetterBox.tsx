import { Colors } from '@/constants/Colors';
import { StyleSheet, View, Text } from 'react-native';

type LetterBoxProps = {
    character: string
}

export default function LetterBox(props: LetterBoxProps) {
    const { character } = props
    return (
        <View style={styles.container}>
            <Text style={styles.title}>{character}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        height: 40,
        width: 40,
        backgroundColor: Colors.secondary,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 8
    },
    title: {
        fontSize: 18,
        fontWeight: 800,
        fontStyle: 'normal',
        color: Colors.primary
    },

});
