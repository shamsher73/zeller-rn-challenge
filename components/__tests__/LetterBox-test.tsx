import { render } from '@testing-library/react-native';
import LetterBox from '../LetterBox';

describe('<LetterBox />', () => {

    it('should render the character when passed as a prop', () => {
        const { getByText } = render(<LetterBox character="A" />);
        expect(getByText("A")).toBeTruthy();
    });

    it('should render empty character when passed as an empty string prop', () => {
        const { getByText } = render(<LetterBox character="" />);
        expect(getByText("")).toBeTruthy();
    });

    test('snapshot', () => {
        const tree = render(<LetterBox character={"S"} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
