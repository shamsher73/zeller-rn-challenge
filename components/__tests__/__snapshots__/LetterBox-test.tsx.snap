// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LetterBox /> snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e8f2fb",
      "borderRadius": 8,
      "height": 40,
      "justifyContent": "center",
      "width": 40,
    }
  }
>
  <Text
    style={
      {
        "color": "#0370ce",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": 800,
      }
    }
  >
    S
  </Text>
</View>
`;
